### PR TITLE
Feature/6 add builder image

### PR DESCRIPTION
**Pull request by blancharda**
This PR reorganizes the folder structure for clarity and adds a builder image.
Users will no longer need to pull external dependencies, or build the c library in order to run the example service.

### CODE REVIEW

File: examples/matching/go/src/mdtf-public/rally2-matching-system/docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
+FROM golang:1.11.2 as builder
+
+#add the source
+RUN mkdir /build
```
[the remark this answers is not in the document]
Comment (blancharda) [reply examples/matching/go/src/mdtf-public/rally2-matching-system/docker/Dockerfile]: Valid point.

File: examples/matching/go/src/mdtf-public/rally2-matching-system/docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+#move the models onto the GOPATH
+RUN mkdir -p $GOPATH/src/mdtf-public/rally2-matching-system
+RUN mv models $GOPATH/src/mdtf-public/rally2-matching-system/
+
+#build the c implementation
+RUN make -C /build/c
+
+#install dependencies
+RUN go get github.com/disintegration/imaging
```
[the remark this answers is not in the document]
Comment (blancharda) [reply examples/matching/go/src/mdtf-public/rally2-matching-system/docker/Dockerfile]: I prefer not to check in 3rd party dependencies... but I'll compromise with dep.

File: examples/matching/go/src/mdtf-public/rally2-matching-system/docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+#build the go wrapper
+RUN GOOS=linux GOARCH=amd64 go build -o ./rally2-matching-system .
+
+#####################
+#build a clean image
+FROM ubuntu:16.04
```
[the remark this answers is not in the document]
Comment (blancharda) [reply examples/matching/go/src/mdtf-public/rally2-matching-system/docker/Dockerfile]: Minimum working example ;)

Check back soon.

File: rally2-matching-system/go-example/README.md [hunk 4]
```diff
@@ -1,7 +1,19 @@
...
 
-This folder contains a simple application, written in [The Go Programming Language](https://golang.org/).  This application implements the [Rally Matching System API](https://github.com/TheMdTF/mdtf-public/blob/master/api/rally2-matching-system/README.md).  Creating an application that implements this API is required for matching system providers to participate in the [2019 Biometric Technology Rally](https://mdtf.org/Rally2019).  Finished applications should be packaged and saved using [Docker](https://www.docker.com/).  Scripts that package and save this application are in the `rally2-matching-system/docker` folder.
+## Overview
+This folder contains a simple application, written in [The Go Programming Language](https://golang.org/).  This application implements the [Rally Matching System API](https://github.com/TheMdTF/mdtf-public/blob/master/api/rally2-matching-system/README.md).  Creating an application that implements this API is required for matching system providers to participate in the [2019 Biometric Technology Rally](https://mdtf.org/Rally2019).  Finished applications should be packaged and saved using [Docker](https://www.docker.com/).
```
Comment (JHowix) [inline rally2-matching-system/go-example/README.md]: the link to the Rally Matching System API will be invalid once this is PRed

File: rally2-matching-system/go-example/README.md [hunk 5]
```diff
@@ -1,7 +1,19 @@
...
 
 It is not required that matching system providers implement the matching system API in Go, as the application in this folder does.  Matching system providers are free to use whatever technology stack they choose, as long as the final docker container conforms to the API.  Scripts that send API conformant requests to a user provided endpoint are provided in the `/api/rally2-matching-system/test` folder.
```
Comment (JHowix) [inline rally2-matching-system/go-example/README.md]: update the /api/rally2-matching-system/test path

File: rally2-matching-system/go-example/README.md [hunk 6]
```diff
@@ -1,7 +1,19 @@
...
 
-To run the sample application using Go, set your `GOPATH` to `mdtf-public/go`.  The only external dependencies for this application are the [Go Disintegration Imaging](https://godoc.org/github.com/disintegration/imaging) package, which can be installed via `go get https://github.com/disintegration/imaging`.  The application can then be run via `go run main.go`.  It can be compiled via `go build -o ./rally2-mathcing-system`.  Once the application has been built, it can be packaged using the scripts in `rally2-matching-system/docker` and tested using the scripts in `/api/rally2-matching-system/test`.
+## Running and Testing via Docker
+The sample application can be packaged and run using the scripts in `./docker` and tested using the scripts in `<mdtf-public>/api/rally2-matching-system/test`.
```
Comment (JHowix) [inline rally2-matching-system/go-example/README.md]: update this api path too?

File: rally2-matching-system/go-example/README.md [hunk 7]
```diff
@@ -1,7 +1,19 @@
...
+
+The script `./docker/save-docker-image.sh` is an example which outputs the docker image in the format we will expect for delivery to the MdTF.
+
+## Manually Building and Running
+To run the sample application using Go:
+ * `go get github.com/mdtf-public/rally2-matching-system/example`
```
Comment (JHowix) [inline rally2-matching-system/go-example/README.md]: I think this path is wrong?

File: rally2-matching-system/go-example/README.md [hunk 8]
```diff
@@ -1,7 +1,19 @@
...
 
-To run the sample application using Go, set your `GOPATH` to `mdtf-public/go`.  The only external dependencies for this application are the [Go Disintegration Imaging](https://godoc.org/github.com/disintegration/imaging) package, which can be installed via `go get https://github.com/disintegration/imaging`.  The application can then be run via `go run main.go`.  It can be compiled via `go build -o ./rally2-mathcing-system`.  Once the application has been built, it can be packaged using the scripts in `rally2-matching-system/docker` and tested using the scripts in `/api/rally2-matching-system/test`.
+## Running and Testing via Docker
+The sample application can be packaged and run using the scripts in `./docker` and tested using the scripts in `<mdtf-public>/api/rally2-matching-system/test`.
+
+The script `./docker/save-docker-image.sh` is an example which outputs the docker image in the format we will expect for delivery to the MdTF.
+
+## Manually Building and Running
```
Comment (JHowix) [inline rally2-matching-system/go-example/README.md]: I think this is going to be confusing to people.  So they don't need to clone the repo, but they can clone the repo or they can just run this command?

**Pull request merged**